### ISSUE TITLE
Special characters

### ISSUE THREAD

**Issue by kennetherland** [AlexPoint/OpenNlp#30]
I am experiencing word splits in weird places when the sentence includes characters like commas, colons, semi-colons, and slashes.

Example:
"As a Hydra Store Visitor, I want to see the latest version."

Becomes:
"As a Hydra Store Vis i t or , I want to see the latest version."

I noticed during TokenizePositions, the last token in the the first phrase ("Visitor,") does not pass the AlphaNumeric.IsMatch(token) because the token has an ending character , i.e. a comma, so the model evaluator takes over and splits up the string in strange ways.  Am I doing something wrong?

Please advise.

**Comment by AlexPoint** [comment]
I'll have a look at your fix quickly and get back to you once the solution has been integrated on the master branch.
Thanks for your help on this!

**Comment by AlexPoint** [comment]
When I run the default tokenizer with the models provided, I get the expected results:
![image](https://user-images.githubusercontent.com/1026807/134808263-3c9ed774-bb82-431c-aed7-52c7c071a1aa.png)

Can you tell me which tokenizer and which models you are using to reproduce this issue?

**Comment by kennetherland** [comment]
Sorry Alex, that I did not see your last email to me.  I should have clarified my goal in the beginning, which was the fact that I was trying to figure out how to map the sub-parses back to the original text locations.  You are right in that the tokenizer tokenizes correctly, but I had to create a few properties to the Parse class and write a routine to analyze the spacing correctly.

I would do a pull request of my code but I went a bit rogue, which would break the port from Java.  I am okay with the closing of the issue.  Thank you for looking into it.
